File: src/Compra.java
import java.time.LocalDate;
import java.util.ArrayList;

public class Compra {
    Cliente cliente;
    ArrayList<Carrito> productos;
    LocalDate fechaCompra;
    public Compra(Cliente cliente,ArrayList<Carrito> productos,LocalDate fechaCompra){
        this.cliente=cliente;
        this.productos=productos;
        this.fechaCompra=fechaCompra;
    }
    public Cliente getCliente() {
        return cliente;
    }

    public LocalDate getFechaCompra() {
        return fechaCompra;
    }
    public ArrayList<Carrito> getProductos(){
        return productos;
    }
    public void mostrarProductos(){
        int i=1;
        System.out.println("\tCant\tProducto");
        for (Carrito carrito:productos){
            System.out.println("\t"+carrito.getCantidad()+"\t"+carrito.getProducto().obtenerInformacion());
            i++;
        }
    }
    public double totalCompra(){
        double total=0;
        for (Carrito carrito:productos){
            total+=carrito.subtotal();
        }
        return total;
    }
    public void mostrarCompra() {
        System.out.println("Cliente: " + cliente.mostrarInfoConFecha());
        System.out.println("Fecha de compra: " + fechaCompra);
        mostrarProductos();
        System.out.println("total: "+totalCompra());
    }
}

File: src/Compra.py
from datetime import date

class Compra:
    def __init__(self, cliente, productos, fecha_compra):
        self.cliente = cliente
        self.productos = productos
        self.fecha_compra = fecha_compra

    def get_cliente(self):
        return self.cliente

    def get_fecha_compra(self):
        return self.fecha_compra

    def get_productos(self):
        return self.productos

    def mostrar_productos(self):
        print("\tCant\tProducto")
        for carrito in self.productos:
            print("\t{}\t{}".format(carrito.get_cantidad(), carrito.get_producto().obtener_informacion()))

    def total_compra(self):
        total = 0
        for carrito in self.productos:
            total += carrito.subtotal()
        return total

    def mostrar_compra(self):
        print("Cliente: " + self.cliente.mostrar_info_con_fecha())
        print("Fecha de compra: " + str(self.fecha_compra))
        self.mostrar_productos()
        print("Total: " + str(self.total_compra()))